grails-app/controllers/org/pih/warehouse/product/CategoryController.groovy

```diff
@@ -53,24 +56,29 @@ class CategoryController {
             categoryInstance.properties = params
         }
 
-        if (!categoryInstance.hasErrors() && categoryInstance.save()) {
+        try {
+            categoryGormService.save(categoryInstance)
             flash.message = "${warehouse.message(code: 'category.saved.message', arg: [format.category(category: categoryInstance).decodeHTML()])}"
             redirect(action: "tree", model: [rootCategory: productService.getRootCategory()])
-        } else {
+        } catch (Exception e) {
```

Comment:
was there a reason for this `try/catch` block, which we agreed we should avoid?
Can't we just refactor the if, not to use `categoryInstance.save()`, but `categoryGormService.save(categoryInstance)`?

Moreover - see the comment below